### ISSUE TITLE
Plugin chokes on anonymous function as rest callback

### ISSUE THREAD

**Issue by fiskhandlarn** [bonny/WordPress-Simple-History#141]
If you add this example code to your site:
```
add_action('rest_api_init', function () {
    if (function_exists('register_rest_route')) {
        $methods = ['GET', 'POST'];
        foreach (['GET', 'POST'] as $method) {
            register_rest_route('test-route', 'test', [
                'methods' => $method,
                'callback' => function ($data) {
                    return new WP_REST_Response([]);
                },
                'args' => [
                    'referer'
                ],
            ]);
        }
    }
});
```

and go to `/wp-json/test-route/test` you get this error:

`Fatal error: Uncaught Error: Cannot use object of type Closure as array in .../simple-history/loggers/Plugin_Redirection.php:101`

Your rest filter should support Closures as well (since they are valid according to the [WordPress REST specs](https://developer.wordpress.org/rest-api/extending-the-rest-api/adding-custom-endpoints/#arguments)).

**Comment by bonny** [comment]
Ah, thanks! I'm still a bit noob when it comes to the api and the callbacks...

**Comment by fiskhandlarn** [comment]
@bonny No worries, I'm no expert on the rest api either :)

**Comment by bonny** [comment]
Should be fixed in 2.23.
